src/main/java/it/gov/pagopa/payhub/activities/service/debtposition/custom/fine/DebtPositionFineProcessor.java

```diff
@@ -0,0 +1,60 @@
+package it.gov.pagopa.payhub.activities.service.debtposition.custom.fine;
+
+import it.gov.pagopa.payhub.activities.dto.debtposition.HandleFineDebtPositionResult;
+import it.gov.pagopa.payhub.activities.dto.debtposition.syncwfconfig.FineWfExecutionConfig;
+import it.gov.pagopa.pu.debtposition.dto.generated.DebtPositionDTO;
+import it.gov.pagopa.pu.debtposition.dto.generated.InstallmentStatus;
+import it.gov.pagopa.pu.debtposition.dto.generated.PaymentOptionStatus;
+import it.gov.pagopa.pu.debtposition.dto.generated.PaymentOptionTypeEnum;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+@Lazy
+@Service
+public class DebtPositionFineProcessor {
+
+    private final DebtPositionFineNotificationDateProcessor dateProcessor;
+
+    public DebtPositionFineProcessor(DebtPositionFineNotificationDateProcessor dateProcessor) {
+        this.dateProcessor = dateProcessor;
+    }
+
+    public HandleFineDebtPositionResult processFine(DebtPositionDTO debtPositionDTO, FineWfExecutionConfig executionParams) {
+        HandleFineDebtPositionResult handleFineDebtPositionResult = dateProcessor.processNotificationDate(debtPositionDTO, executionParams);
+
+        if (!handleFineDebtPositionResult.getReductionEndDate().isBefore(OffsetDateTime.now())) {
```

Comment:
it's the same, but I found it more readable wrote as inverse
ps: we are on the reductionPeriod even if there is not a reduction end date
```suggestion

        // It we are currenctly on the reduction period
        if (handleFineDebtPositionResult.getReductionEndDate()== null || handleFineDebtPositionResult.getReductionEndDate().isAfter(OffsetDateTime.now())) {
```